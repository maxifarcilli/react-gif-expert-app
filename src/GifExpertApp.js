import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ( { defaultCategories = [] } ) => {
    
    const [categories, setCategories] = useState( defaultCategories );
    

    return (
        <>

            <h1>GifExpertApp</h1>
            <AddCategory addCategory={ setCategories } />
            <hr />
            
            <section>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category } />
                    ))
                }
            </section>
            
        </>
    );
};

export default GifExpertApp;


/**
 * Componente - GifExpertApp
 *  <h2>GifExpertApp</h2>
 *  <hr />
 */

