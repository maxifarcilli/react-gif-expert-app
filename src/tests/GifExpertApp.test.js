import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas sobre el componente principal <GifExpertApp />', () => {
    
    test('Debe de mostrar el componente correctamente ', () => {
        
        const wrapper = shallow(
            <GifExpertApp />
        );

        expect( wrapper ).toMatchSnapshot();

    });
    
    test('Debe de mostrar una lista de categorias', () => {
       
        const categories = ['Football', 'Messi'];

        const wrapper = shallow(
            <GifExpertApp defaultCategories={ categories } />
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );


    });
    
});
