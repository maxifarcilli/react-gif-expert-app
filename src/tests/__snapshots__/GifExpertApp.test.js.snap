// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente principal <GifExpertApp /> Debe de mostrar el componente correctamente  1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <section />
</Fragment>
`;

exports[`Pruebas sobre el componente principal <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <section>
    <GifGrid
      category="Football"
      key="Football"
    />
    <GifGrid
      category="Messi"
      key="Messi"
    />
  </section>
</Fragment>
`;
